### PR TITLE
Zuhang fix team management alert if user already exists

### PR DESCRIPTION
**Pull request by zuhang2000**
# Description
(PRIORITY MEDIUM) Jae: Fix Team Management alert if user already exists
a. Other Links -> Teams -> Choose Team → Members → Add Existing
b. If you try to add an existing member, it says “Please choose a valid user.” It should say, “Member is already a part of this team.”

## Main changes explained:
- Address a bug in file src/components/Teams/MembersAutoComplete.jsx.
- Update file src/components/Teams/TeamMembersPopup.jsx. Adding a new state called "duplicateUserAlert" to detect duplicate member.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as admin user
4. go to dashboard→ Other Links→ Teams→Members
5. Add new members or existing members

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123051008/4cb27aa9-2b9d-48d5-82af-8c5e3047c396

### CODE REVIEW

**Review by winghojackyli — DISMISSED** [review]
Hi @zuhang2000 , I've tested this PR and it works as intended. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/d998a80e-28e0-4218-a4f3-d464fdcbfbd8

**Comment by Arsamat** [comment]
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67019552/e8634d71-0673-4191-a389-e6062fe56b55

@zuhang2000 Tested the functionality. Works as expected!

**Comment by luisarevalo21** [comment]
Hello @zuhang2000 
I tried to test your branch but I am missing the links in the Navbar. 
This was a bug that started last Monday, any suggestions on fixing it?
When it happened to my branch, I had to reset my entire development environment. 


Screenshot 2023-08-27 at 9 06 20 PM

**Comment by tsunami776** [comment]
I had the same issue as @luisarevalo21. The links in the navigation are missing for me, and FE dev branch has those links. 

PR Review #1185

**Review by harshidad — DISMISSED** [review]
Changes looks Good!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/b118dc71-9746-4954-b7eb-e2b501674164

**Review by yurisokolovicz — COMMENTED** [review]
Dear @zuhang2000 

I am also experiencing the issue reported in the error section regarding the absence of "other links". I have tried clearing the cache and following the normal procedures, but it still is not working.


Screenshot 2023-08-28 232155

**Comment by luisarevalo21** [comment]
> I had the same issue as @luisarevalo21. The links in the navigation are missing for me, and FE dev branch has those links.
> 
> UPDATE: I made it work by checkout an old commit with detached HEAD on dd8809c for BE. The features work as described in the PR.
> 
> PR Review #1185
> 
>  PR.Review.1185.mp4

Hello can you explain exactly, how you resolved the missing links in the navbar?

**Comment by tsunami776** [comment]
@luisarevalo21 First of all, I do not think my method is a fix to this. It only made it work to test it. And here is how I did it through GitHub Desktop as shown in the screenshot. I just checkout an old back end commit from 10 days ago. That was all I did.

Screenshot 2023-08-29

**Review by navneeeth — DISMISSED** [review]
LGTM! Initially faced the issue with 'Other Links' not showing up. Removing the cache and reloading the page did not help. Followed @tsunami776 's steps to checkout to the 'dd8809c' commit for the BE and it worked as a fix.

The PR works as intended. Good job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44264834/58464e18-bf12-4972-9861-c7cd3052619e

**Comment by AnishPandita** [comment]
Hey, the PR works well. Tried adding new user and already existing user.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111321945/5d4c66e6-a4b8-4d1e-8d64-9e3eb6aaa4d1

**Review by DavidC0126 — DISMISSED** [review]
Hi @zuhang2000 I tested your PR and it works as intended.

already_exist_alert

**Review by Haoj1 — DISMISSED** [review]
@zuhang2000 Good job! Alert when add existing members. Refresh did not fix the other link disappeared issues. I try @tsunami776 solutions and switch to previous BE commit works for me. Perhaps some bugs are merged into current BE dev.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/56e4a1dc-9965-4ae0-8e23-c3fd40f44b18

**Review by vinaykumarh — COMMENTED** [review]
I have reviewed the PR.
As mentioned by earlier reviewers the nav links are missing along with logout button on profile page. I can see conflicts in the branch, please resolve it.
As for the PR it is working as expected. One thing I observed is the username is getting cleared in new fix whereas it is persist in search bar in old functionality. I believe keeping it in the search bar will be useful.
Attached screenshot!

Before:
image

After:
image

Nav missing:
image

**Review by Indujaks — DISMISSED** [review]
Tested your PR, it works as expected, I tried adding a existing team member and the error is shown!
Screenshot 2023-08-30 at 11 40 28 AM

**Review by yurisokolovicz — DISMISSED** [review]
Dear @zuhang2000 

I'm also encountering the problem mentioned in the error section, where the "other links" are missing. I attempted to clear the cache (using "Empty Cache and Hard Reload") and followed the usual steps, but the issue persists. Even creating another owner account didn't fix the problem (this approach worked for other PRs). 


Screenshot 2023-08-30 195524

**Comment by Jiyuanxia** [comment]
Hi, @zuhang2000 , I met the same issue that the tab for other links is missing.

Screenshot 2023-08-30 172845

**Review by Chuehleo — DISMISSED** [review]
Everything works as expected, great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/122568562/32e868ac-05d9-4c65-8e92-9b989de7c686

**Review by gabi-lima — DISMISSED** [review]
Well done.
image

**Comment by zuhang2000** [comment]
> Dear @zuhang2000
> 
> I'm also encountering the problem mentioned in the error section, where the "other links" are missing. I attempted to clear the cache (using "Empty Cache and Hard Reload") and followed the usual steps, but the issue persists. Even creating another owner account didn't fix the problem (this approach worked for other PRs).
> 
> Screenshot 2023-08-30 195524

I would it say it's probably the problem with the most recent back-end development branch. I also encounter similar issue when reviewing other PRs. Could you please test with older back-end? I believe this problem is not related to this PR.

**Review by scoopstroop92 — DISMISSED** [review]
Works as expected . Good job 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/89441435/5f48c25c-bac3-4611-8ea2-6def047fbcb8

**Review by GaryB93 — CHANGES_REQUESTED** [review]
I too am unable to access the Other Links to test this PR. I know this has been fixed so merging current development branch to this branch should fix this issue.

Screenshot 2023-09-07 at 3 55 02 PM

Screenshot 2023-09-07 at 4 02 26 PM

**Comment by zuhang2000** [comment]
> I too am unable to access the Other Links to test this PR. I know this has been fixed so merging current development branch to this branch should fix this issue.
> 
> Screenshot 2023-09-07 at 3 55 02 PM
> 
> Screenshot 2023-09-07 at 4 02 26 PM

I have solved merge conflicts. And I have merged current dev branch. The problem is now fixed. Please check on that! Thank you so much!

**Review by GaryB93 — DISMISSED** [review]
Hi @zuhang2000, great work on implementing the error message upon trying to add a member to a team they are already a member of. I tested this under an Admin and Owner role and it works flawlessly.
Screenshot 2023-09-14 at 12 54 43 PM

**Comment by Shubhu0** [comment]
Tested PR, error messages being displayed correctly.

**Review by Sssky-18 — APPROVED** [review]
Hi @zuhang2000 , just tested this PR and the feature worked fine for me. I tried to add an existing member, and the warning text displayed correctly. 
PR#1185

**Comment by Shubhu0** [comment]
Hi @zuhang2000 reviewed the PR and it is working as intended. 

Created and added a new member
image

Tested for adding and already added member
image

**Comment by RheaWu1212** [comment]
works really well except my network was slowwww

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109626741/7a012c78-d347-49f5-833a-9cc7ae1d4df5

**Review by Bhagyashree444 — APPROVED** [review]
Works as expected. Good job!

Screenshot (195)

Screenshot (194)

**Review by JeffLi117 — DISMISSED** [review]
Changes work as expected, good job!

Added initial user JEFFTESTCAPS Li:
image

Attempting to add same user again:
image

The error message:
image

**Review by yi-lin-1234 — DISMISSED** [review]
Hello, I tested your PR and works as intended. Great work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92185695/79b6544f-31b1-4c4a-bb77-cb87d8b866db

**Comment by Lin1404** [comment]
Hi @zuhang2000 tested the PR and it is working as intended.
PR1185

**Review by Lin1404 — DISMISSED** [review]

**Review by StrawberryCalpico — DISMISSED** [review]
I reviewed your PR, it works as the description. Good work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114443664/7ab05a89-b4e7-46b1-9dac-e8086a18b420

**Comment by kavya151** [comment]
I have examined the PR, and it functions correctly when a team member's name is chosen from the dropdown. However, when I type the full member name, it displays "Please choose a valid user" instead of "The member is already part of this team."
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/61841929/6b649f7e-5f4a-4616-988f-e708e6be13bd

**Review by cvtqx — CHANGES_REQUESTED** [review]
I've faced the same issue as @kavya151 . When choosing a name from the dropown, the new message appears as described. However if we  type the full name of an existing team member manually, the message is different:

Screen Shot 2023-11-07 at 10 40 27 AM

Here is a video of testing:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/e5ca1180-77d5-4a5b-93ee-dcceb40b60dd

File: src/components/Teams/TeamMembersPopup.jsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ export const TeamMembersPopup = React.memo(props => {
   const [selectedUser, onSelectUser] = useState(undefined);
   const [isValidUser, onValidation] = useState(true);
```
Comment (cvtqx) [inline src/components/Teams/TeamMembersPopup.jsx:17]: Can these 2 state update functions (lines 16 and 17) names be updated to follow the convention? The convention for naming the state update functions is to use a prefix like "set" followed by the state variable name.

**Review by cvtqx — APPROVED** [review]
Thanks for updating the code to follow the convention. As for my previous comment, I think it should be addressed in a new PR, I will let Jae know.

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**